staging/src/k8s.io/legacy-cloud-providers/aws/aws.go

```diff
@@ -1303,14 +1295,43 @@ func newAWSCloud(cfg CloudConfig, awsServices Services) (*Cloud, error) {
 		}
 	}
 
-	// Register regions, in particular for ECR credentials
-	once.Do(func() {
-		recognizeWellKnownRegions()
-	})
-
 	return awsCloud, nil
 }
 
+// isRegionValid accepts an AWS region name and returns if the region is a
+// valid region known to the AWS SDK. Considers the region returned from the
+// EC2 metadata service to be a valid region as it's only available on a host
+// running in a valid AWS region.
+func isRegionValid(region string, metadata EC2Metadata) bool {
```

Comment:
Would it be worthwhile to have unit tests that runs against the set of well known regions? Might be useful to catch updates in the SDK that we weren't expected. It may not catch new regions but it will at least validate against the ones we know are used

Reply:
I'm happy to write the test but I'm not sure what value they would add. The SDK regions are generated from an internal system and this code-path is really just used to validate that if a user provided a region override that it's valid. Previously this was also used to generate credential providers but that's no longer the case.

Reply:
Ah, so `endpoints.DefaultPartitions` is checking metadata for all existing regions?

Reply:
I had the impression that the regions list was still hardcoded but at least it's hard coded into the SDK. If it's hard-coded in the SDK than unit tests to validate that the SDK allows for (at least) the existing well known regions could be useful to make sure we're not regressing on this behavior when we bump the SDK versions to support new regoisn

Reply:
Well the list in the SDK is generated from a data management system so it's not really hard-coded. But I've added a test to validate this functionality anyhow. WDYT?

Reply:
Looks great! Can we also add 1 error case (invalid region, not in metadata)?

Reply:
Added another test case. LGTY?